src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -146,11 +146,25 @@ def __init__(
 
     @property
     def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
+        from snowflake.snowpark._internal.analyzer.analyzer import ARRAY_BIND_THRESHOLD
+
+        literal_complexity = len(self.data) * len(self.output)
```

Comment:
@sfc-gh-aalam How about provide a property or method in SnowflakeValue like is_large_local_data, which does check len(self.data) * len(self.output) < ARRAY_BIND_THRESHOLD, and make thishttps://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/analyzer/analyzer.py#L960 also calls into this? we can also offer data size property that also does calculation len(self.data) * len(self.output), so that we centralizes all those logic inside SnowflakeValues, and if one day, we want to make changes to this part, we don't have to change different places